sdk/storage/Azure.Storage.DataMovement.Files.Shares/src/ShareFileStorageResource.cs

```diff
@@ -347,6 +347,60 @@ protected override StorageResourceCheckpointDetails GetDestinationCheckpointDeta
                 isDirectoryMetadataSet: _options?._isDirectoryMetadataSet ?? false,
                 directoryMetadata: _options?.DirectoryMetadata);
         }
+
+        protected override async Task ValidateTransferAsync(StorageResource sourceResource, CancellationToken cancellationToken = default)
```

Comment:
Hmm unfortunate we need to write this twice. Could we move essentially the whole function to some helper at least so it can be shared between single and container resources?